src/main/java/qna/service/QnAService.java

```diff
@@ -34,25 +34,11 @@ public Question findQuestionById(Long id) {
 
     @Transactional
     public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
+        MyQuestions myQuestions = MyQuestions.of(question, loginUser);
+        myQuestions.delete(deleteHistories);
```

Comment:
delete 메서드가 삭제한 내역을 반환하는건 어떨까요?
```suggestion
        DeleteHistories deleteHistories = question.delete(deleteHistories);
```